lib/Remotery.c

```diff
@@ -5757,7 +5757,10 @@ static rmtError SampleThreadsLoop(rmtThread* rmt_thread)
             // While the thread is suspended take the chance to check for samples trees that may never complete
             // Because SuspendThread on Windows is an async request, this needs to be placed at a point where the request completes
             // Calling GetThreadContext will ensure the request is completed so this stall check is placed after that
-            CheckForStallingSamples(&stalling_sample_tree, thread_profiler, sample_time_us);
+            if (RMT_ERROR_NONE != CheckForStallingSamples(&stalling_sample_tree, thread_profiler, sample_time_us))
+            {
+                SampleTree_Destructor(&stalling_sample_tree);
```

Comment:
I don't believe this is correct:

1. `stalling_sample_tree` is allocated via `SampleTree_Copy`.
2. This *aliases* the `allocator` copied from the source tree.
3. As such; `stalling_sample_tree` does not own its `allocator` pointer.
4. `SampleTree_Destructor` explicitly calls `Delete` on the `allocator` pointer, which it doesn't own.

I think the best course of action here is to call `FreeSamples` instead of `SampleTree_Destructor`.

Reply:
I see. This actually never happens (in my tests) but since `CheckForStallingSamples` may return with an error I beleive releasing the samples tree is the right thing to do here.

Should the upper bit releasing the `root` be using the `FreeSamples` as well (instead of the `ObjectAllocator_Free` directly)?

Reply:
`FreeSamples` is recursive: it will flatten an entire tree of samples and then walk that list and free them each.

Given that you're only releasing the root, your call to `ObjectAllocator_Free` is already achieving its goal.

However, calling `FreeSamples` there is probably a good idea also as the API can be cleaned up at a later date.

Reply:
Great. I change it to use `FreeSamples` in there as well then.